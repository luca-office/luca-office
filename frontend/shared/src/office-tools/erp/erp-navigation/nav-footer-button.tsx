import {css} from "@emotion/react"
import React from "react"
import {Icon, Text} from "../../../components"
import {IconName} from "../../../enums"
import {
  backgroundColorBright,
  borderRadius,
  CustomStyle,
  deepShadow,
  Flex,
  iconBrightColor,
  primaryColor,
  spacing,
  spacingHuge,
  spacingSmall,
  TextSize
} from "../../../styles"

export interface NavFooterButtonProps extends CustomStyle {
  readonly onClick: () => void
  readonly iconName: IconName
  readonly label: string
}

export const NavFooterButton: React.FC<NavFooterButtonProps> = ({onClick, iconName, label, customStyles}) => {
  return (
    <div css={customStyles}>
      <div css={styles.button} onClick={onClick}>
        <div css={styles.iconContainer}>
          <Icon color={iconBrightColor} name={iconName} />
        </div>
        <Text size={TextSize.Medium}>{label}</Text>
      </div>
    </div>
  )
}

const styles = {
  button: css(Flex.row, {
    height: spacingHuge,
    background: backgroundColorBright,
    boxShadow: deepShadow,
    borderRadius: borderRadius,
    cursor: "pointer"
  }),
  iconContainer: css(Flex.row, {
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: primaryColor,
    width: spacingHuge,
    height: spacingHuge,
    borderRadius: spacing(borderRadius, 0, 0, borderRadius),
    marginRight: spacingSmall
  })
}
