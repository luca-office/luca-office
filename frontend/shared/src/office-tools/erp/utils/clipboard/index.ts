export * from "./clipboard"
export * from "./common"
export * from "./referencing-keys-clipboard"
