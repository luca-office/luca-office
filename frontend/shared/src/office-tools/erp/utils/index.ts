export * from "./auto-complete"
export * from "./clipboard"
export * from "./common"
export * from "./erp-entity-key"
export * from "./erp-navigation-entry-label"
export * from "./input"
export * from "./label"
export * from "./number"
export * from "./referencing-keys"
export * from "./required"
export * from "./scenario-erp-entity"
export * from "./template"
