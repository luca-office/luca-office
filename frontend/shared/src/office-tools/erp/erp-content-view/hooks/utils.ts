import {ErpEntity} from "../../../../models"
import {ErpTableEntity, ErpTableRowData} from "../../erp-table"

export const erpEntityToTableEntity = (erpEntity: ErpEntity): ErpTableEntity =>
  Object.keys(erpEntity).reduce((acc, cur) => {
    return ["sampleCompanyId", "binaryFile"].includes(cur) ? acc : {...acc, [cur]: (erpEntity as any)[cur]}
  }, {} as ErpTableEntity)

export const erpEntityToTableRow = (erpEntity: ErpEntity, index: number): ErpTableRowData => ({
  entityId: erpEntity.id,
  index,
  entity: erpEntityToTableEntity(erpEntity)
})
