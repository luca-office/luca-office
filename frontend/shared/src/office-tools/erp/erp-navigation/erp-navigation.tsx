import {css} from "@emotion/react"
import * as React from "react"
import {Checkbox, Icon, TableOfContentsContainer, TableOfContentsEntry, Text} from "../../../components"
import {IconName, NodeType} from "../../../enums"
import {BaseNode, ErpEntry} from "../../../models"
import {CustomStyle, Flex, flex1, fontColorLight, spacingSmall, TextSize} from "../../../styles"
import {useLucaTranslation} from "../../../translations"
import {Option} from "../../../utils"
import {useErpNavigation} from "./hooks/use-erp-navigation"
import {NavFooterButton} from "./nav-footer-button"

export interface ErpNavigationProps extends CustomStyle {
  readonly navigationEntries: ErpEntry[]
  readonly onEntrySelected: (node: BaseNode) => void
  readonly isReadonly?: boolean
  readonly onImportClicked?: () => void
  readonly onExportClicked?: () => void
  readonly onChangeCollapseState?: (node: BaseNode, isCollapsed: boolean) => void
  readonly onSelectTable?: (node: BaseNode) => void
  readonly selectedNode: Option<string>
}

export const ErpNavigation: React.FC<ErpNavigationProps> = ({
  navigationEntries,
  isReadonly = false,
  customStyles,
  onEntrySelected,
  onImportClicked,
  onExportClicked,
  onChangeCollapseState,
  selectedNode: selectedNodeId
}) => {
  const {t} = useLucaTranslation()

  const {
    selectedNode,
    setSelectedNode,
    hideLockedEntries,
    toggleHideLockedEntries,
    baseNodes,
    expandedNodeIds
  } = useErpNavigation(navigationEntries, selectedNodeId)

  const handleSelectNode = (nodeOption: Option<BaseNode>) => {
    nodeOption.forEach(node => node.type === NodeType.Table && onEntrySelected(node))

    setSelectedNode(nodeOption.map(node => node.id))
  }

  const hideLockedEntriesCheckbox = (
    <div key={"erp-navigation-checkbox"} className="erp-navigation-checkbox" css={Flex.row}>
      <Text size={TextSize.Medium}>{t("erp__navigation_hide_placeholders")}</Text>
      <Checkbox onChange={toggleHideLockedEntries} checked={hideLockedEntries} customStyles={styles.checkbox} />
    </div>
  )

  const footerButtons = (
    <div css={styles.footerButtons}>
      <NavFooterButton
        customStyles={styles.importButton}
        onClick={() => onImportClicked?.()}
        iconName={IconName.Upload}
        label={t("erp__navigation_import_excel_file")}
      />
      <NavFooterButton
        onClick={() => onExportClicked?.()}
        iconName={IconName.Download}
        label={t("erp__navigation_export_excel_file")}
      />
    </div>
  )

  const renderErpEntry = (node: BaseNode, onClick: () => void) => {
    return (
      <div key={node.id} onClick={onClick} css={styles.entry}>
        <div css={Flex.row}>
          <Icon
            customStyles={styles.entryIcon}
            name={node.type === NodeType.Directory ? IconName.Folder : IconName.Table}
          />
          <Text size={TextSize.Small}>{node.name}</Text>
        </div>
        {node.isLocked && <Icon color={fontColorLight} name={IconName.LockClosed} />}
      </div>
    )
  }

  return (
    <TableOfContentsContainer
      customCardStyles={customStyles}
      customFooterStyles={styles.footer}
      title={t("erp__title")}
      headerButtons={isReadonly ? [] : [hideLockedEntriesCheckbox]}
      actionFooter={isReadonly ? undefined : footerButtons}>
      {baseNodes.map(node => (
        <TableOfContentsEntry
          key={node.id}
          node={node}
          selectNode={handleSelectNode}
          selectedNode={selectedNode}
          renderCustomNodeContent={renderErpEntry}
          isCollapsible={true}
          onChangeCollapseState={onChangeCollapseState}
          expandedNodeIds={expandedNodeIds}
        />
      ))}
    </TableOfContentsContainer>
  )
}

const styles = {
  entry: css(Flex.row, {
    flex: flex1,
    justifyContent: "space-between"
  }),
  checkbox: css({
    marginLeft: spacingSmall
  }),
  entryIcon: css({
    marginRight: spacingSmall
  }),
  footer: css({
    height: "auto"
  }),
  footerButtons: css(Flex.column),
  importButton: css({
    marginBottom: spacingSmall
  })
}
