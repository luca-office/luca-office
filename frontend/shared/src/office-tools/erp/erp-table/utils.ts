import {Sorting} from "../../../enums"
import {
  DeliveryStatus,
  EmploymentMode,
  FamilyStatus,
  PaymentStatus,
  Salutation
} from "../../../graphql/generated/globalTypes"
import {backgroundColorBright, foreignKeyColor, primaryKeyColor} from "../../../styles"
import {LucaTFunction} from "../../../translations"
import {convertCentsToEuro, formatDate, parseDateString} from "../../../utils"
import {
  getDeliveryStatusLabel,
  getEmploymentModeLabel,
  getFamilyStatusLabel,
  getPaymentStatusLabel,
  getSalutationLabel
} from "../utils"
import {ErpTableColumnType, ErpTableContentType} from "./erp-table"
import {ErpTableSortConfig} from "./erp-table-header"
import {ErpTableRowProps} from "./erp-table-row"

export const backgroundColorByColumnType = (type: ErpTableColumnType) => {
  switch (type) {
    case ErpTableColumnType.PrimaryKey:
      return primaryKeyColor
    case ErpTableColumnType.ForeignKey:
      return foreignKeyColor
    case ErpTableColumnType.Default:
    case ErpTableColumnType.Selector:
    default:
      return backgroundColorBright
  }
}

export const nextSortDirection = (currentSorting: Sorting) =>
  currentSorting === Sorting.None
    ? Sorting.Ascending
    : currentSorting === Sorting.Ascending
    ? Sorting.Descending
    : Sorting.None

type SortRowProps = Omit<ErpTableRowProps, "index" | "vStyle">

export const compareFnBySortConfig = (sortConfig: ErpTableSortConfig) => {
  switch (sortConfig.contentType) {
    case ErpTableContentType.Number:
    case ErpTableContentType.Currency:
      return (a: SortRowProps, b: SortRowProps) =>
        sortConfig.sorting === Sorting.Ascending
          ? ((a.entity[sortConfig.propertyKey] as unknown) as number) -
            ((b.entity[sortConfig.propertyKey] as unknown) as number)
          : ((b.entity[sortConfig.propertyKey] as unknown) as number) -
            ((a.entity[sortConfig.propertyKey] as unknown) as number)
    case ErpTableContentType.Date:
      return (a: SortRowProps, b: SortRowProps) =>
        sortConfig.sorting === Sorting.Ascending
          ? new Date(((a.entity[sortConfig.propertyKey] ?? "") as unknown) as string).getTime() -
            new Date(((b.entity[sortConfig.propertyKey] ?? "") as unknown) as string).getTime()
          : new Date(((b.entity[sortConfig.propertyKey] ?? "") as unknown) as string).getTime() -
            new Date(((a.entity[sortConfig.propertyKey] ?? "") as unknown) as string).getTime()
    case ErpTableContentType.TextArray:
      return (a: SortRowProps, b: SortRowProps) =>
        sortConfig.sorting === Sorting.Ascending
          ? (((a.entity[sortConfig.propertyKey] ?? []) as unknown) as Array<string>)
              .join(",")
              .localeCompare((((b.entity[sortConfig.propertyKey] ?? []) as unknown) as Array<string>).join(","))
          : (((b.entity[sortConfig.propertyKey] ?? []) as unknown) as Array<string>)
              .join(",")
              .localeCompare((((a.entity[sortConfig.propertyKey] ?? []) as unknown) as Array<string>).join(","))
    case ErpTableContentType.Text:
    case ErpTableContentType.BinaryFile:
    default:
      return (a: SortRowProps, b: SortRowProps) => {
        return sortConfig.sorting === Sorting.Ascending
          ? (((a.entity[sortConfig.propertyKey] ?? "") as unknown) as string).localeCompare(
              ((b.entity[sortConfig.propertyKey] ?? "") as unknown) as string
            )
          : (((b.entity[sortConfig.propertyKey] ?? "") as unknown) as string).localeCompare(
              ((a.entity[sortConfig.propertyKey] ?? "") as unknown) as string
            )
      }
  }
}

export const formatCellValue = (
  contentType: ErpTableContentType,
  value: string | Array<string> | number | null,
  t: LucaTFunction
) => {
  switch (contentType) {
    case ErpTableContentType.BinaryFile:
      return value ? t("erp__table_attachment_available") : t("erp__table_attachment_not_available")
    case ErpTableContentType.Currency: {
      if (value === null) {
        return ""
      } else {
        const numberValue = Number(value)
        return `${numberValue > 0 ? convertCentsToEuro(numberValue) : numberValue} â‚¬`
      }
    }
    case ErpTableContentType.Date:
      return value ? formatDate(parseDateString(String(value))) : ""
    case ErpTableContentType.Salutation:
      return value ? getSalutationLabel(t, value as Salutation) : ""
    case ErpTableContentType.EmploymentMode:
      return value ? getEmploymentModeLabel(t, value as EmploymentMode) : ""
    case ErpTableContentType.FamilyStatus:
      return value ? getFamilyStatusLabel(t, value as FamilyStatus) : ""
    case ErpTableContentType.PaymentStatus:
      return value ? getPaymentStatusLabel(t, value as PaymentStatus) : ""
    case ErpTableContentType.DeliveryStatus:
      return value ? getDeliveryStatusLabel(t, value as DeliveryStatus) : ""
    case ErpTableContentType.TextArray:
      return value ? (value as Array<string>).join(", ") : ""
    case ErpTableContentType.Text:
    case ErpTableContentType.Number:
    default:
      return `${value ?? ""}`
  }
}
