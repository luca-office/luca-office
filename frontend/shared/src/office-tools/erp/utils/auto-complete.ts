import {ErpType} from "../../../enums"
import {ErpEntityByType, ErpKeyEntity} from "../../../models"
import {find} from "../../../utils"
import {UseErpForeignKeysHook} from "../hooks"

export const getAutoCompleteList = <T extends ErpType>(
  autoCompleteLists: TypeOf<UseErpForeignKeysHook, "foreignKeys">,
  foreignKeys: ErpKeyEntity<T>[],
  key: keyof ErpEntityByType<T>
): string[] =>
  find(foreignKey => foreignKey.key === key, foreignKeys)
    .map(foreignKey => autoCompleteLists[foreignKey.type] ?? [])
    .getOrElse([])
