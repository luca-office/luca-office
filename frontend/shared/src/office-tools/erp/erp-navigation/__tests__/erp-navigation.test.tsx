import {shallow} from "enzyme"
import * as React from "react"
import {create} from "react-test-renderer"
import {fakeTranslate} from "../../../../../tests/utils/translate-mock"
import {TableOfContentsContainer, TableOfContentsEntry} from "../../../../components"
import {Option} from "../../../../utils"
import {ErpNavigation, ErpNavigationProps} from "../erp-navigation"
import {NavFooterButton} from "../nav-footer-button"
import {getStaticErpStructure} from "../static-erp-structure"

const navigationEntries = getStaticErpStructure("Test Company", fakeTranslate)

const defaultProps: ErpNavigationProps = {
  navigationEntries,
  onEntrySelected: jest.fn(),
  selectedNode: Option.none()
}

const getComponent = (props?: Partial<ErpNavigationProps>) => <ErpNavigation {...{...defaultProps, ...props}} />

describe("erp-navigation", () => {
  it("renders correctly with default props", () => {
    const component = getComponent()
    const tree = create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("renders correctly in readonly mode", () => {
    const component = getComponent({isReadonly: true})
    const tree = create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("has correct structure with default props", () => {
    const component = shallow(getComponent())

    expect(component.find(TableOfContentsContainer)).toHaveLength(1)
    expect(component.find(TableOfContentsEntry)).toHaveLength(navigationEntries.length) // shallow

    const tocContainer = component.find(TableOfContentsContainer).first()
    expect(tocContainer.dive().find(".erp-navigation-checkbox")).toHaveLength(1)
    expect(tocContainer.dive().find(NavFooterButton)).toHaveLength(2)
  })

  it("has correct structure in readonly mode", () => {
    const component = shallow(getComponent({isReadonly: true}))

    expect(component.find(TableOfContentsContainer)).toHaveLength(1)
    expect(component.find(TableOfContentsEntry)).toHaveLength(navigationEntries.length) // shallow

    const tocContainer = component.find(TableOfContentsContainer).first()
    expect(tocContainer.dive().find(".erp-navigation-checkbox")).toHaveLength(0)
    expect(tocContainer.dive().find(".erp-navigation-import-button")).toHaveLength(0)
  })
})
