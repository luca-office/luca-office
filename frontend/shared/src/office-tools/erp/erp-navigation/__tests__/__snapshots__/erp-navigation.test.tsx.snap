// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`erp-navigation renders correctly in readonly mode 1`] = `
<div
  className="css-1tz1azl-Card"
>
  <div
    className="card-header css-15zaw2t-CardHeader"
  >
    <div
      className="css-mv0otu-TableOfContentsContainer"
    >
      <div
        className="css-1f6p2vj-Heading"
      >
        erp__title
      </div>
    </div>
    <div
      className="css-t84163-TableOfContentsContainer"
    />
  </div>
  <div
    className="toc-body css-awggpr-TableOfContentsContainer"
  >
    <div
      className="book-chapters-list css-1n23nny-TableOfContentsContainer"
    >
      <div
        className="css-18xeog2-TableOfContentsContainer"
      >
        <div
          className="toc-entry css-12opz0x-TableOfContentsEntry"
        >
          <div
            className="css-1ldtu9f-Icon"
            dangerouslySetInnerHTML={
              {
                "__html": {},
              }
            }
            onClick={[Function]}
          />
          <div
            className="css-1ympzk2-TableOfContentsEntry"
            onClick={[Function]}
          >
            <div
              className="css-ya6sn-TableOfContentsEntry"
            >
              <div
                className="css-1lrzij5-Icon"
                dangerouslySetInnerHTML={
                  {
                    "__html": {},
                  }
                }
              />
              <div
                className="css-1uyodls-Text"
              >
                Test Company
              </div>
            </div>
            <div
              className="css-1p4ztl7-Icon"
              dangerouslySetInnerHTML={
                {
                  "__html": {},
                }
              }
            />
          </div>
        </div>
        <div
          className="toc-entry css-12opz0x-TableOfContentsEntry"
        >
          <div
            className="css-1ldtu9f-Icon"
            dangerouslySetInnerHTML={
              {
                "__html": {},
              }
            }
            onClick={[Function]}
          />
          <div
            className="css-1ympzk2-TableOfContentsEntry"
            onClick={[Function]}
          >
            <div
              className="css-ya6sn-TableOfContentsEntry"
            >
              <div
                className="css-1lrzij5-Icon"
                dangerouslySetInnerHTML={
                  {
                    "__html": {},
                  }
                }
              />
              <div
                className="css-1uyodls-Text"
              >
                erp__navigation_label_office
              </div>
            </div>
            <div
              className="css-1p4ztl7-Icon"
              dangerouslySetInnerHTML={
                {
                  "__html": {},
                }
              }
            />
          </div>
        </div>
        <div
          className="toc-entry css-12opz0x-TableOfContentsEntry"
        >
          <div
            className="css-1ldtu9f-Icon"
            dangerouslySetInnerHTML={
              {
                "__html": {},
              }
            }
            onClick={[Function]}
          />
          <div
            className="css-1ympzk2-TableOfContentsEntry"
            onClick={[Function]}
          >
            <div
              className="css-ya6sn-TableOfContentsEntry"
            >
              <div
                className="css-1lrzij5-Icon"
                dangerouslySetInnerHTML={
                  {
                    "__html": {},
                  }
                }
              />
              <div
                className="css-1uyodls-Text"
              >
                erp__navigation_label_application_spanning_components
              </div>
            </div>
            <div
              className="css-1p4ztl7-Icon"
              dangerouslySetInnerHTML={
                {
                  "__html": {},
                }
              }
            />
          </div>
        </div>
        <div
          className="toc-entry css-12opz0x-TableOfContentsEntry"
        >
          <div
            className="css-1ldtu9f-Icon"
            dangerouslySetInnerHTML={
              {
                "__html": {},
              }
            }
            onClick={[Function]}
          />
          <div
            className="css-1ympzk2-TableOfContentsEntry"
            onClick={[Function]}
          >
            <div
              className="css-ya6sn-TableOfContentsEntry"
            >
              <div
                className="css-1lrzij5-Icon"
                dangerouslySetInnerHTML={
                  {
                    "__html": {},
                  }
                }
              />
              <div
                className="css-1uyodls-Text"
              >
                erp__navigation_label_logistics
              </div>
            </div>
          </div>
        </div>
        <div
          className="toc-entry css-12opz0x-TableOfContentsEntry"
        >
          <div
            className="css-1ldtu9f-Icon"
            dangerouslySetInnerHTML={
              {
                "__html": {},
              }
            }
            onClick={[Function]}
          />
          <div
            className="css-1ympzk2-TableOfContentsEntry"
            onClick={[Function]}
          >
            <div
              className="css-ya6sn-TableOfContentsEntry"
            >
              <div
                className="css-1lrzij5-Icon"
                dangerouslySetInnerHTML={
                  {
                    "__html": {},
                  }
                }
              />
              <div
                className="css-1uyodls-Text"
              >
                erp__navigation_label_accounting
              </div>
            </div>
            <div
              className="css-1p4ztl7-Icon"
              dangerouslySetInnerHTML={
                {
                  "__html": {},
                }
              }
            />
          </div>
        </div>
        <div
          className="toc-entry css-12opz0x-TableOfContentsEntry"
        >
          <div
            className="css-1ldtu9f-Icon"
            dangerouslySetInnerHTML={
              {
                "__html": {},
              }
            }
            onClick={[Function]}
          />
          <div
            className="css-1ympzk2-TableOfContentsEntry"
            onClick={[Function]}
          >
            <div
              className="css-ya6sn-TableOfContentsEntry"
            >
              <div
                className="css-1lrzij5-Icon"
                dangerouslySetInnerHTML={
                  {
                    "__html": {},
                  }
                }
              />
              <div
                className="css-1uyodls-Text"
              >
                erp__navigation_label_staff
              </div>
            </div>
          </div>
        </div>
        <div
          className="toc-entry css-12opz0x-TableOfContentsEntry"
        >
          <div
            className="css-1ldtu9f-Icon"
            dangerouslySetInnerHTML={
              {
                "__html": {},
              }
            }
            onClick={[Function]}
          />
          <div
            className="css-1ympzk2-TableOfContentsEntry"
            onClick={[Function]}
          >
            <div
              className="css-ya6sn-TableOfContentsEntry"
            >
              <div
                className="css-1lrzij5-Icon"
                dangerouslySetInnerHTML={
                  {
                    "__html": {},
                  }
                }
              />
              <div
                className="css-1uyodls-Text"
              >
                erp__navigation_label_info_systems
              </div>
            </div>
            <div
              className="css-1p4ztl7-Icon"
              dangerouslySetInnerHTML={
                {
                  "__html": {},
                }
              }
            />
          </div>
        </div>
        <div
          className="toc-entry css-12opz0x-TableOfContentsEntry"
        >
          <div
            className="css-1ldtu9f-Icon"
            dangerouslySetInnerHTML={
              {
                "__html": {},
              }
            }
            onClick={[Function]}
          />
          <div
            className="css-1ympzk2-TableOfContentsEntry"
            onClick={[Function]}
          >
            <div
              className="css-ya6sn-TableOfContentsEntry"
            >
              <div
                className="css-1lrzij5-Icon"
                dangerouslySetInnerHTML={
                  {
                    "__html": {},
                  }
                }
              />
              <div
                className="css-1uyodls-Text"
              >
                erp__navigation_label_tools
              </div>
            </div>
            <div
              className="css-1p4ztl7-Icon"
              dangerouslySetInnerHTML={
                {
                  "__html": {},
                }
              }
            />
          </div>
        </div>
        <div
          className="toc-entry css-12opz0x-TableOfContentsEntry"
        >
          <div
            className="css-1ldtu9f-Icon"
            dangerouslySetInnerHTML={
              {
                "__html": {},
              }
            }
            onClick={[Function]}
          />
          <div
            className="css-1ympzk2-TableOfContentsEntry"
            onClick={[Function]}
          >
            <div
              className="css-ya6sn-TableOfContentsEntry"
            >
              <div
                className="css-1lrzij5-Icon"
                dangerouslySetInnerHTML={
                  {
                    "__html": {},
                  }
                }
              />
              <div
                className="css-1uyodls-Text"
              >
                erp__navigation_label_web_client_ui_framework
              </div>
            </div>
            <div
              className="css-1p4ztl7-Icon"
              dangerouslySetInnerHTML={
                {
                  "__html": {},
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="card-footer css-1gfqsqp-CardFooter"
  />
</div>
`;

exports[`erp-navigation renders correctly with default props 1`] = `
<div
  className="css-1tz1azl-Card"
>
  <div
    className="card-header css-15zaw2t-CardHeader"
  >
    <div
      className="css-mv0otu-TableOfContentsContainer"
    >
      <div
        className="css-1f6p2vj-Heading"
      >
        erp__title
      </div>
    </div>
    <div
      className="css-t84163-TableOfContentsContainer"
    >
      <div
        className="erp-navigation-checkbox css-j81lzu-ErpNavigation"
      >
        <div
          className="css-17zbm7y-Text"
        >
          erp__navigation_hide_placeholders
        </div>
        <div
          className="css-12kot9v-Checkbox"
        >
          <div
            className="css-1k20ryc-Checkbox"
          >
            <div
              className="css-kb4bik-Checkbox"
            >
              <input
                checked={false}
                className="css-1gxpz9h-Checkbox"
                disabled={false}
                onChange={[Function]}
                readOnly={false}
                role="checkbox"
                type="checkbox"
              />
              <div
                className="css-xgozj0-Checkbox"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="toc-body css-awggpr-TableOfContentsContainer"
  >
    <div
      className="book-chapters-list css-1n23nny-TableOfContentsContainer"
    >
      <div
        className="css-18xeog2-TableOfContentsContainer"
      >
        <div
          className="toc-entry css-12opz0x-TableOfContentsEntry"
        >
          <div
            className="css-1ldtu9f-Icon"
            dangerouslySetInnerHTML={
              {
                "__html": {},
              }
            }
            onClick={[Function]}
          />
          <div
            className="css-1ympzk2-TableOfContentsEntry"
            onClick={[Function]}
          >
            <div
              className="css-ya6sn-TableOfContentsEntry"
            >
              <div
                className="css-1lrzij5-Icon"
                dangerouslySetInnerHTML={
                  {
                    "__html": {},
                  }
                }
              />
              <div
                className="css-1uyodls-Text"
              >
                Test Company
              </div>
            </div>
            <div
              className="css-1p4ztl7-Icon"
              dangerouslySetInnerHTML={
                {
                  "__html": {},
                }
              }
            />
          </div>
        </div>
        <div
          className="toc-entry css-12opz0x-TableOfContentsEntry"
        >
          <div
            className="css-1ldtu9f-Icon"
            dangerouslySetInnerHTML={
              {
                "__html": {},
              }
            }
            onClick={[Function]}
          />
          <div
            className="css-1ympzk2-TableOfContentsEntry"
            onClick={[Function]}
          >
            <div
              className="css-ya6sn-TableOfContentsEntry"
            >
              <div
                className="css-1lrzij5-Icon"
                dangerouslySetInnerHTML={
                  {
                    "__html": {},
                  }
                }
              />
              <div
                className="css-1uyodls-Text"
              >
                erp__navigation_label_office
              </div>
            </div>
            <div
              className="css-1p4ztl7-Icon"
              dangerouslySetInnerHTML={
                {
                  "__html": {},
                }
              }
            />
          </div>
        </div>
        <div
          className="toc-entry css-12opz0x-TableOfContentsEntry"
        >
          <div
            className="css-1ldtu9f-Icon"
            dangerouslySetInnerHTML={
              {
                "__html": {},
              }
            }
            onClick={[Function]}
          />
          <div
            className="css-1ympzk2-TableOfContentsEntry"
            onClick={[Function]}
          >
            <div
              className="css-ya6sn-TableOfContentsEntry"
            >
              <div
                className="css-1lrzij5-Icon"
                dangerouslySetInnerHTML={
                  {
                    "__html": {},
                  }
                }
              />
              <div
                className="css-1uyodls-Text"
              >
                erp__navigation_label_application_spanning_components
              </div>
            </div>
            <div
              className="css-1p4ztl7-Icon"
              dangerouslySetInnerHTML={
                {
                  "__html": {},
                }
              }
            />
          </div>
        </div>
        <div
          className="toc-entry css-12opz0x-TableOfContentsEntry"
        >
          <div
            className="css-1ldtu9f-Icon"
            dangerouslySetInnerHTML={
              {
                "__html": {},
              }
            }
            onClick={[Function]}
          />
          <div
            className="css-1ympzk2-TableOfContentsEntry"
            onClick={[Function]}
          >
            <div
              className="css-ya6sn-TableOfContentsEntry"
            >
              <div
                className="css-1lrzij5-Icon"
                dangerouslySetInnerHTML={
                  {
                    "__html": {},
                  }
                }
              />
              <div
                className="css-1uyodls-Text"
              >
                erp__navigation_label_logistics
              </div>
            </div>
          </div>
        </div>
        <div
          className="toc-entry css-12opz0x-TableOfContentsEntry"
        >
          <div
            className="css-1ldtu9f-Icon"
            dangerouslySetInnerHTML={
              {
                "__html": {},
              }
            }
            onClick={[Function]}
          />
          <div
            className="css-1ympzk2-TableOfContentsEntry"
            onClick={[Function]}
          >
            <div
              className="css-ya6sn-TableOfContentsEntry"
            >
              <div
                className="css-1lrzij5-Icon"
                dangerouslySetInnerHTML={
                  {
                    "__html": {},
                  }
                }
              />
              <div
                className="css-1uyodls-Text"
              >
                erp__navigation_label_accounting
              </div>
            </div>
            <div
              className="css-1p4ztl7-Icon"
              dangerouslySetInnerHTML={
                {
                  "__html": {},
                }
              }
            />
          </div>
        </div>
        <div
          className="toc-entry css-12opz0x-TableOfContentsEntry"
        >
          <div
            className="css-1ldtu9f-Icon"
            dangerouslySetInnerHTML={
              {
                "__html": {},
              }
            }
            onClick={[Function]}
          />
          <div
            className="css-1ympzk2-TableOfContentsEntry"
            onClick={[Function]}
          >
            <div
              className="css-ya6sn-TableOfContentsEntry"
            >
              <div
                className="css-1lrzij5-Icon"
                dangerouslySetInnerHTML={
                  {
                    "__html": {},
                  }
                }
              />
              <div
                className="css-1uyodls-Text"
              >
                erp__navigation_label_staff
              </div>
            </div>
          </div>
        </div>
        <div
          className="toc-entry css-12opz0x-TableOfContentsEntry"
        >
          <div
            className="css-1ldtu9f-Icon"
            dangerouslySetInnerHTML={
              {
                "__html": {},
              }
            }
            onClick={[Function]}
          />
          <div
            className="css-1ympzk2-TableOfContentsEntry"
            onClick={[Function]}
          >
            <div
              className="css-ya6sn-TableOfContentsEntry"
            >
              <div
                className="css-1lrzij5-Icon"
                dangerouslySetInnerHTML={
                  {
                    "__html": {},
                  }
                }
              />
              <div
                className="css-1uyodls-Text"
              >
                erp__navigation_label_info_systems
              </div>
            </div>
            <div
              className="css-1p4ztl7-Icon"
              dangerouslySetInnerHTML={
                {
                  "__html": {},
                }
              }
            />
          </div>
        </div>
        <div
          className="toc-entry css-12opz0x-TableOfContentsEntry"
        >
          <div
            className="css-1ldtu9f-Icon"
            dangerouslySetInnerHTML={
              {
                "__html": {},
              }
            }
            onClick={[Function]}
          />
          <div
            className="css-1ympzk2-TableOfContentsEntry"
            onClick={[Function]}
          >
            <div
              className="css-ya6sn-TableOfContentsEntry"
            >
              <div
                className="css-1lrzij5-Icon"
                dangerouslySetInnerHTML={
                  {
                    "__html": {},
                  }
                }
              />
              <div
                className="css-1uyodls-Text"
              >
                erp__navigation_label_tools
              </div>
            </div>
            <div
              className="css-1p4ztl7-Icon"
              dangerouslySetInnerHTML={
                {
                  "__html": {},
                }
              }
            />
          </div>
        </div>
        <div
          className="toc-entry css-12opz0x-TableOfContentsEntry"
        >
          <div
            className="css-1ldtu9f-Icon"
            dangerouslySetInnerHTML={
              {
                "__html": {},
              }
            }
            onClick={[Function]}
          />
          <div
            className="css-1ympzk2-TableOfContentsEntry"
            onClick={[Function]}
          >
            <div
              className="css-ya6sn-TableOfContentsEntry"
            >
              <div
                className="css-1lrzij5-Icon"
                dangerouslySetInnerHTML={
                  {
                    "__html": {},
                  }
                }
              />
              <div
                className="css-1uyodls-Text"
              >
                erp__navigation_label_web_client_ui_framework
              </div>
            </div>
            <div
              className="css-1p4ztl7-Icon"
              dangerouslySetInnerHTML={
                {
                  "__html": {},
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="card-footer css-1gfqsqp-CardFooter"
  >
    <div
      className="css-3lsagf-ErpNavigation"
    >
      <div
        className="css-1koh34k-NavFooterButton"
      >
        <div
          className="css-4vcym2-NavFooterButton"
          onClick={[Function]}
        >
          <div
            className="css-18owz2d-NavFooterButton"
          >
            <div
              className="css-1pfqrtn-Icon"
              dangerouslySetInnerHTML={
                {
                  "__html": {},
                }
              }
            />
          </div>
          <div
            className="css-17zbm7y-Text"
          >
            erp__navigation_import_excel_file
          </div>
        </div>
      </div>
      <div
        className="css-0"
      >
        <div
          className="css-4vcym2-NavFooterButton"
          onClick={[Function]}
        >
          <div
            className="css-18owz2d-NavFooterButton"
          >
            <div
              className="css-1pfqrtn-Icon"
              dangerouslySetInnerHTML={
                {
                  "__html": {},
                }
              }
            />
          </div>
          <div
            className="css-17zbm7y-Text"
          >
            erp__navigation_export_excel_file
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
